### ISSUE TITLE
SmugMugAuthorize.AuthorizeSmugMug apparently requires console

### ISSUE THREAD

**Issue by frogger1** [AlexGhiondea/SmugMug.NET#29]
Great API - keep up the good work!  I seem to have run into an issue where running the AuthorizeSmugMug method requires a console window to press enter once the web browser is done.  Is there a way to call this pragmatically given the username and password?  Opening a prompt for the user is a bit cumbersome - and deferring to a web browser is not preferred.

**Comment by AlexGhiondea** [comment]
I have been thinking about that and I think the flow for oAuth can be simplified, but it would require you to run the application as admin the first time. This is a limitation of how the oAuth flow works -- it was not designed to work well with non-web applications.

Here is how I got it to work for other projects:
- You provide the API key and Client secret
- Using those, a browser is launched and the user is authorized. You need to use localhost as the return uri
- The application starts an `HttpListener` for the localhost address that the web browser will redirect to
- The application will pull out the code from the redirected call and use it as the token

I will take a look at this and see if I can create a way to do this.

**Comment by AlexGhiondea** [comment]
Argh -- looks like the 1.3.0 API does not support redirecting the browser after the authentication has happened.

The reason why you need to wait until someone interacts with the console is because it takes time for the user to login on the site and authenticate the application. And there is no way of knowing when the access token is available.

I guess one could wait and periodically check and see if they can exchange the request token with the access token. I am not sure if that would work.

**Comment by frogger1** [comment]
I ended up workaround this by modifying/extending the demo code to only perform the auth and then store the encrypted credentials locally... So basically, the first time my app runs, I call the command line utility as a new process.   A little inconvenient, but workable (for now).

I did find that the XDocument does not get encrypted/decrypted properly in the sample code - but by switching to encrypting a JSON string the process works fine.  Thanks!

**Comment by AlexGhiondea** [comment]
Yeah -- that would work! As long as the token is retrieved once (and every time it expires, if needed) then the application wouldn't need to see the console piece at all :).

I will take a look at the encryption part -- I think I fixed it in the 2.0 version of the API, but I don't remember porting it back to 1.3.
